### PR TITLE
Add a read file function to utils.js

### PR DESCRIPTION
**Pull request by kirstyellis**
We have been using blockly with Erle's robot_blockly and found that some of the blocks of code for our more complex blocks were quite large so rather than adding the code into the generators/python js files, we created a folder inside generators/python called scripts where we add python files containing the code for that each block. We added the readFile function to utils.js so that we could then read in these python files. Thought it might be a handy feature to add.

Here is an example :

https://github.com/shadow-robot/blockly/blob/master/generators/python/arm.js#L11

### CODE REVIEW

**Comment by kirstyellis** [comment]
I signed it!

**Comment by carlosperate** [comment]
XMLHttpRequest is not really used at all in Blockly, which permits it's use without having a server providing the Blockly application. While an interesting feature, adding it would break that advantage from Blockly.

**Comment by NeilFraser** [comment]
I'm not seeing why this should be part of Blockly since Blockly does not call it.  It should just be part of your host application.

It is also functionality that's provided by most libraries.  Closure (used by Blockly) already has this in the goog.net.XhrIo package.  jQuery already has this as $.ajax.

**Comment by kirstyellis** [comment]
Thanks for your response @carlosperate, I will close the PR.
